activity_browser/ui/tables/models/base.py

```diff
@@ -64,14 +65,23 @@ def data(self, index, role=Qt.DisplayRole):
                 value = value.item()
             elif isinstance(value, tuple):
                 value = str(value)
-            elif isinstance(value, datetime.datetime) and role != 'sorting':
+            elif isinstance(value, datetime.datetime) and (Qt.DisplayRole or Qt.ToolTipRole):
                 tz = datetime.datetime.now(datetime.timezone.utc).astimezone()
                 time_shift = - tz.utcoffset().total_seconds()
-                value = arrow.get(value).shift(seconds=time_shift).humanize()
+                if role == Qt.ToolTipRole:
+                    value = arrow.get(value).shift(seconds=time_shift).format('YYYY-MM-DD HH:mm:ss')
+                    tt_date_flag = True
+                elif role == Qt.DisplayRole:
+                    value = arrow.get(value).shift(seconds=time_shift).humanize()
+
+        # immediately return value in case of DisplayRole or sorting
+        if role == Qt.DisplayRole or role == 'sorting':
+            return value
+
+        # in case of ToolTipRole and date, always show the full date
+        if tt_date_flag and role == Qt.ToolTipRole:
+            return value
```

Comment:
Can't you just return the value where you've set the flag? That would clear the code up nicely

Reply:
In that case you wouldn't need the flag to begin with.

Reply:
> Can't you just return the value where you've set the flag? That would clear the code up nicely

That's what I had earlier, but thought this may be more consistent as there's no returns in other places